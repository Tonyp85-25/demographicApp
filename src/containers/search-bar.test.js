// import axios from 'axios';
// import {fetchCountries} from './index'
import React from 'react';
import { Provider } from 'react-redux';
import {act} from 'react-dom/test-utils'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {render, screen} from '@testing-library/react';
import SearchContainer from './search-bar';

// jest.mock('axios');

// test('should get countries', async () =>{
//     const countries = [{name: "Groland", currencyCode: "GRD$",flag: "myflag", code: "GRD"}]
//     const resp = {data: countries}
//     axios.get.mockValue(resp)
//     return fetchCountries().then(d)
// })
const mockStore = configureStore([]);
describe('components > SearchContainer', () => {
    let component;
    let store;
  
    beforeEach(() => {
      store = mockStore({
        countryReducer: {
          countries: [{name: "Groland", currencyCode: "GRD$",flag: "myflag", code: "GRD"}],
        },
        
      });
   
      component = renderer.create(
        <Provider store={store}>
          <SearchContainer/>
        </Provider>
     );
    });
    it('renders without crashing', () => {
        const div = document.createElement('div');
        act(()=>{
          render(<Provider store={store}><SearchContainer/></Provider>, div);
        })
        
    });
    it('should render with given state from Redux store', () => {
      expect(component.toJSON()).toMatchSnapshot();
    });

});