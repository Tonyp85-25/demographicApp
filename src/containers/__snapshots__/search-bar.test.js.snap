// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > SearchContainer should render with given state from Redux store 1`] = `
<form>
  <div
    className="inline-block relative w-64"
  >
    <select
      className="block appearance-none w-full rounded bg-teal-300 border border-gray-200 shadow-lg text-indigo-700 hover:text-white py-3 px-4 pr-8 rounded leading-tight focus:outline-none focus:bg-white focus:border-gray-500 focus:text-black"
      onChange={[Function]}
    >
      <option
        className=""
        value="GRD"
      >
        Groland
      </option>
    </select>
    <div
      className="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700"
    >
      <svg
        className="fill-current h-4 w-4"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
        />
      </svg>
    </div>
  </div>
</form>
`;
